### PR TITLE
improve hyperparameters documentation

### PR DESCRIPTION
**Pull request by pmeier**
@jbueltemeier you can look at the built documentation by clicking on "Details" for the readthedocs.org check below.

### CODE REVIEW

File: docs/source/api/pystiche_papers.li_wand_2016.rst [hunk 1]
```diff
@@ -27,31 +27,149 @@
...
+-------------------------
+
+Unfortunately, the the behavior in the reference implementation differ from what is
+described in the paper. The following parts are affected:
+
+- :class:`~pystiche_papers.li_wand_2016.FeatureReconstructionOperator`
+- :class:`~pystiche_papers.li_wand_2016.MRFOperator`
+- :class:`~pystiche_papers.li_wand_2016.TotalVariationOperator`
+- :class:`~pystiche_papers.li_wand_2016.target_transforms`
+
```
Comment (jbueltemeier) [inline docs/source/api/pystiche_papers.li_wand_2016.rst:42]: The pyramid is also influenced by this parameter.

> num_steps = 100 if impl_params else 200
num_levels = 3 if impl_params else None

File: docs/source/api/pystiche_papers.li_wand_2016.rst [hunk 2]
```diff
@@ -27,31 +27,149 @@
...
++------------------+-----------------+-----------+
+| Parameter        | ``impl_params``             |
++                  +-----------------+-----------+
+|                  | ``True``        | ``False`` |
++==================+=================+===========+
+| ``layer``        | ``"relu4_2"``               |
++------------------+-----------------+-----------+
+| ``score_weight`` | ``2e1``         | ``1e0``   |
++------------------+-----------------+-----------+
```
Comment (jbueltemeier) [inline docs/source/api/pystiche_papers.li_wand_2016.rst:63]: What about the loss_reduction?

> self.loss_reduction = "mean" if impl_params else "sum"

File: docs/source/api/pystiche_papers.li_wand_2016.rst [hunk 3]
```diff
@@ -27,31 +27,149 @@
...
+| ``layers``        | ``("relu3_1", "relu4_1")``  |
++-------------------+-----------------+-----------+
+| ``layer_weights`` | ``"sum"``                   |
++-------------------+-----------------+-----------+
+| ``patch_size``    | ``3``                       |
++-------------------+-----------------+-----------+
+| ``stride``        | ``2``           | ``1``     |
++-------------------+-----------------+-----------+
+| ``score_weight``  | ``1e-4``        | ``1e0``   |
++-------------------+-----------------+-----------+
```
Comment (jbueltemeier) [inline docs/source/api/pystiche_papers.li_wand_2016.rst:100]: `score_correction_factor` is missing. 

However, this is currently being discussed in [#228](https://github.com/pmeier/pystiche_papers/issues/228).

**Comment by pmeier** [comment]
I basically split the changes into two categories:

1. behavioral changes and
2. hyper parameter changes.

The former includes "unreasonable" changes (`score_correction_factor`s, `loss_reduction`s other than `"mean"`, ...), while the latter includes "reasonable" changes (`score_weight`, `num_steps`, ...).

All behavioral changes are listed in the `Implementation Parameters` section. In this category I think we should only give the user the ability to use it with or without `impl_params`. Thus, these parameters are not part of the `hyper_parameters()`.

On the other hand, functionality that has no behavioral changes (for example `image_pyramid`) is not listed in the `Implementation Parameters` section.

**Comment by jbueltemeier** [comment]
Okay, I wasn't aware of that, but this is a very good solution. So you can forget my comments above, they only refer to the behavioral changes.

**Comment by pmeier** [comment]
> @jbueltemeier you can look at the built documentation by clicking on "Details" for the readthedocs.org check below.

:wink:

File: docs/source/api/pystiche_papers.ulyanov_et_al_2016.rst [hunk 4]
```diff
@@ -27,106 +27,226 @@
...
+|                  | ``True``             | ``False`` |
++                  +----------------------+-----------+
+|                  | ``ìnstance_norm``    |           |
++                  +----------+-----------+           +
+|                  | ``True`` | ``False`` |           |
++==================+==========+===========+===========+
+| ``layer``        | ``"relu4_2"``                    |
++------------------+----------+-----------+-----------+
+| ``score_weight`` | ``1e0``  | ``6e-1``  | ``1e0``   |
++------------------+----------+-----------+-----------+
```
[the remark this answers is not in the document]
Comment (pmeier) [reply docs/source/api/pystiche_papers.ulyanov_et_al_2016.rst]: Fixed in e51a85931ba26e8de5b77df82be53d7721635add

File: docs/source/api/index.rst [hunk 5]
```diff
@@ -5,9 +5,42 @@ Package reference
...
+Unfortunately, the reference implementation of the original authors often deviates from
+what is described in the paper. To account for this, the ``impl_params`` flag is used.
+It defaults to ``True`` since the parameters in the implementation were probably used
+to generate the results in the paper.
+
+In general, the deviations can be separated into two classes:
+
+1. **Behavioral changes**: These changes often result by misconceptions of the author
+   for how their used framework or library works internally. An example for this the
+   inclusion of a score correction factor of ``1/2`` for quadratic losses in the paper,
```
[the remark this answers is not in the document]
Comment (pmeier) [reply docs/source/api/index.rst]: Fixed in c9521b1a51775f363f7bcfc3a4d9d7fca513f80d

**Review by jbueltemeier — APPROVED** [review]

**Pull request merged**